src/djangooidc/oidc.py

```diff
@@ -183,6 +183,8 @@ def callback(self, unparsed_response, session):
         if authn_response["state"] != session.get("state", None):
             # this most likely means the user's Django session vanished
             logger.error("Received state not the same as expected for %s" % state)
+            if session.get("state", None) is None:
+                raise o_e.NoStateDefined()
```

Comment:
(Q) should this check come before line 183? If authn_response["state"] == None and session.get("state", None) == None, then the "NoStateDefined" error will never get raised.

I am not too familiar with this area so feel free to push back on this